javascript/modules/client/src/client-addressList.ts

```diff
@@ -1,640 +1,61 @@
-  IPluginInstallItem,
-  SortDirection,
-} from "./internal/interfaces/common";
-import {
-  AddressListProposal,
-  AddressListProposalListItem,
-  ExecuteProposalStep,
-  ExecuteProposalStepValue,
   IAddressListPluginInstall,
-  ICanVoteParams,
   IClientAddressList,
-  ICreateProposalParams,
-  IExecuteProposalParams,
-  IPluginSettings,
-  IProposalQueryParams,
-  IVoteProposalParams,
-  ProposalCreationSteps,
-  ProposalCreationStepValue,
-  ProposalMetadata,
-  ProposalSortBy,
-  SubgraphAddressListProposal,
-  SubgraphAddressListProposalListItem,
-  VoteProposalStep,
-  VoteProposalStepValue,
+  IClientAddressListDecoding,
+  IClientAddressListEncoding,
+  IClientAddressListEstimation,
+  IClientAddressListMethods,
 } from "./internal/interfaces/plugins";
-import { delay } from "./internal/temp-mock";
-import {
-  computeProposalStatusFilter,
-  isProposalId,
-  toAddressListProposal,
-  toAddressListProposalListItem,
-} from "./internal/utils/plugins";
-import { isAddress } from "@ethersproject/address";
-import {
-  QueryAddressListProposal,
-  QueryAddressListProposals,
-} from "./internal/graphql-queries/proposal";
-import { QueryAddressListPluginSettings } from "./internal/graphql-queries/settings";
-
-// NOTE: This address needs to be set when the plugin has been published and the ID is known
-const PLUGIN_ID = "0x1234567890123456789012345678901234567890";
+import { ClientAddressListMethods } from "./internal/client-addressList/methods";
+import { ClientAddressListEncoding } from "./internal/client-addressList/encoding";
+import { ClientAddressListDecoding } from "./internal/client-addressList/decoding";
+import { ClientAddressListEstimation } from "./internal/client-addressList/estimation";
+import { IPluginInstallItem } from "./internal/interfaces/common";
+import { ClientCore } from "./internal/core";
 
 /**
  * Provider a generic client with high level methods to manage and interact an Address List Voting plugin installed in a DAO
  */
 export class ClientAddressList extends ClientCore
   implements IClientAddressList {
+  private privateMethods: IClientAddressListMethods;
+  private privateEncoding: IClientAddressListEncoding;
+  private privateDecoding: IClientAddressListDecoding;
+  private privateEstimation: IClientAddressListEstimation;
+
   constructor(context: ContextPlugin) {
     super(context);
+    this.privateMethods = new ClientAddressListMethods(context);
+    this.privateEncoding = new ClientAddressListEncoding(context);
+    this.privateDecoding = new ClientAddressListDecoding(context);
+    this.privateEstimation = new ClientAddressListEstimation(context);
+  }
+  get methods(): IClientAddressListMethods {
+    return this.privateMethods;
+  }
+  get encoding(): IClientAddressListEncoding {
+    return this.privateEncoding;
+  }
+  get decoding(): IClientAddressListDecoding {
+    return this.privateDecoding;
+  }
+  get estimation(): IClientAddressListEstimation {
+    return this.privateEstimation;
```

Comment:
Private variables can still be changed. The proposal below may be an overkill and still not add much protection, just for inspiration:

```suggestion

const privateMethods = Symbol("privateMethods");
const privateEncoding = Symbol("privateEncoding");
const privateDecoding = Symbol("privateDecoding");
const privateEstimation = Symbol("privateEstimation");

/**
 * Provider a generic client with high level methods to manage and interact an Address List Voting plugin installed in a DAO
 */
export class ClientAddressList extends ClientCore
  implements IClientAddressList {
  private [privateMethods]: IClientAddressListMethods = new ClientAddressListMethods(context);
  private [privateEncoding]: IClientAddressListEncoding = new ClientAddressListEncoding(context);
  private [privateDecoding]: IClientAddressListDecoding = new ClientAddressListDecoding(context);
  private [privateEstimation]: IClientAddressListEstimation = new ClientAddressListEstimation(context);

  constructor(context: ContextPlugin) {
    super(context);
  }
  get methods(): IClientAddressListMethods {
    return this[privateMethods];
  }
  get encoding(): IClientAddressListEncoding {
    return this[privateEncoding];
  }
  get decoding(): IClientAddressListDecoding {
    return this[privateDecoding];
  }
  get estimation(): IClientAddressListEstimation {
    return this[privateEstimation];
```